pkg/processor/runtime/golang/runtime.go

```diff
@@ -119,17 +151,58 @@ func (g *golang) callEntrypoint(event nuclio.Event, functionLogger logger.Logger
 		}
 	}()
 
-	// before we call, save timestamp
-	startTime := time.Now()
+	ctx, cancel := context.WithCancel(context.Background())
+	g.cancelEventHandlingChan <- cancel
+	responseChan := make(chan *processingResponse, 1)
+	// Run the function in a goroutine
+	go func() {
+		defer func() {
+			if err := recover(); err != nil {
+				callStack := debug.Stack()
+				functionLogger.WarnWith("Panic caught in event handler",
+					"err",
+					err,
+					"stack",
+					string(callStack))
+			}
+		}()
+		// before we call, save timestamp
+		startTime := time.Now()
+
+		// call entrypoint
+		response, responseErr = g.entrypoint(g.Context, event)
 
-	response, responseErr = g.entrypoint(g.Context, event)
+		// calculate how long it took to invoke the function
+		callDuration := time.Since(startTime)
 
-	// calculate how long it took to invoke the function
-	callDuration := time.Since(startTime)
+		// signal that results
+		responseChan <- &processingResponse{
+			response,
+			responseErr,
+		}
 
-	// add duration to sum
-	g.Statistics.DurationMilliSecondsSum += uint64(callDuration.Nanoseconds() / 1000000)
-	g.Statistics.DurationMilliSecondsCount++
+		// add duration to sum
+		g.Statistics.DurationMilliSecondsSum += uint64(callDuration.Nanoseconds() / 1000000)
+		g.Statistics.DurationMilliSecondsCount++
+	}()
+	select {
+	case result := <-responseChan:
+		select {
+		case cancelEventHandling := <-g.cancelEventHandlingChan:
```

Comment:
You already have the `cancel` function from above, is this only to free-up the channel?

Reply:
@TomerShor yep